packages/beacon-node/src/chain/validation/blobSidecar.ts

```diff
@@ -0,0 +1,154 @@
+      blobIdx: blobSidecar.index,
+    });
+  }
+
+  if (!blsKeyValidate(blobSidecar.kzgCommitment)) {
+    throw new BlobSidecarError(GossipAction.REJECT, {
+      code: BlobSidecarErrorCode.INVALID_KZG,
+      blobIdx: blobSidecar.index,
+    });
+  }
+
+  // verifyKzgCommitmentsAgainstTransactions will need to be done while block verification when block
+  // would also be available
+
+  // [REJECT] The KZG proof is a correctly encoded compressed BLS G1 Point
+  // -- i.e. blsKeyValidate(blobs_sidecar.kzg_aggregated_proof)
+  if (!blsKeyValidate(blobSidecar.kzgProof)) {
+    throw new BlobSidecarError(GossipAction.REJECT, {
+      code: BlobSidecarErrorCode.INVALID_KZG_PROOF,
+      blobIdx: blobSidecar.index,
+    });
+  }
+
+  validateBlobsAndProofs([blobSidecar.kzgCommitment], [blobSidecar.blob], [blobSidecar.kzgProof]);
+}
+
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/beacon-chain.md#validate_blobs_sidecar
+export function validateBlobSidecars(
+  blockSlot: Slot,
+  blockRoot: Root,
+  transactions: bellatrix.Transactions,
+  expectedKzgCommitments: deneb.BlobKzgCommitments,
+  blobSidecars: deneb.BlobSidecars
+): void {
+  // TODO: freetheblobs
+  if (!verifyKzgCommitmentsAgainstTransactions(transactions, expectedKzgCommitments)) {
+    throw new Error("Invalid block transactions and commitments");
+  }
+
+  // assert len(expected_kzg_commitments) == len(blobs)
+  if (expectedKzgCommitments.length !== blobSidecars.length) {
+    throw new Error(
+      `blobSidecars length to commitments length mismatch. Blob length: ${blobSidecars.length}, Expected commitments length ${expectedKzgCommitments.length}`
+    );
+  }
+
+  // No need to verify the aggregate proof of zero blobs. Also c-kzg throws.
+  // https://github.com/dankrad/c-kzg/pull/12/files#r1025851956
+  if (blobSidecars.length > 0) {
+    // Verify the blob slot and root matches
+    const blobs = [];
+    const proofs = [];
+    for (let index = 0; index < blobSidecars.length; index++) {
+      const blobSidecar = blobSidecars[index];
+      if (
+        blobSidecar.slot !== blockSlot ||
+        !byteArrayEquals(blobSidecar.blockRoot, blockRoot) ||
+        blobSidecar.index !== index ||
+        !byteArrayEquals(expectedKzgCommitments[index], blobSidecar.kzgCommitment)
+      ) {
+        throw new Error(
+          `Invalid blob with slot=${blobSidecar.slot} blockRoot=${toHex(blockRoot)} index=${
+            blobSidecar.index
+          } for the block root=${toHex(blockRoot)} slot=${blockSlot} index=${index}`
+        );
+      }
+      blobs.push(blobSidecar.blob);
+      proofs.push(blobSidecar.kzgProof);
+    }
+    validateBlobsAndProofs(expectedKzgCommitments, blobs, proofs);
+  }
+}
+
+function validateBlobsAndProofs(
+  expectedKzgCommitments: deneb.BlobKzgCommitments,
+  blobs: deneb.Blobs,
+  proofs: deneb.KZGProofs
+): void {
+  // assert verify_aggregate_kzg_proof(blobs, expected_kzg_commitments, kzg_aggregated_proof)
```

Comment:
Noting comment re `kzg_aggregated_proof`.

(Its still a draft so I imagine you were on your way to clean these up anyways :) )